tests/comments/test_async_serializer.py

```diff
@@ -0,0 +1,229 @@
+import pytest
+from django.contrib.contenttypes.models import ContentType
+from django.urls import reverse
+
+from adhocracy4.comments.models import Comment
+from tests.apps.questions.models import Question
+
+
+@pytest.mark.django_db
+def test_serializer(apiclient, question, comment_factory, user):
+    question_ct = ContentType.objects.get_for_model(Question)
+    comment_ct = ContentType.objects.get_for_model(Comment)
+    moderator = question.project.moderators.first()
+
+    comment_factory(pk=1, content_object=question, creator=user)
+    comment_factory(pk=2, content_object=question, creator=user,
+                    comment_categories='QUE')
+    comment_factory(pk=3, content_object=question, creator=user,
+                    is_blocked=True)
+    comment_factory(pk=4, content_object=question, creator=user,
+                    is_censored=True)
+    comment_factory(pk=5, content_object=question, creator=user,
+                    is_removed=True)
+    comment_factory(pk=6, content_object=question, creator=user,
+                    is_moderator_marked=True)
+    comment = comment_factory(pk=7, content_object=question, creator=moderator)
+    comment_factory(pk=8, content_object=comment, creator=user)
+
+    apiclient.force_authenticate(user=user)
+    url = reverse(
+        'comments_async-list',
+        kwargs={'content_type': question_ct.pk,
+                'object_pk': question.pk})
+    response = apiclient.get(url)
+
+    comment_data = response.data['results']
+    assert len(comment_data) == 7
+
+    # Testing a plain comment
+    comment_1 = comment_data[6]
+    assert comment_1['id'] == 1
+    assert comment_1['user_name'] == ''  # should be user.username ?!!
+    # In the react_comments_async template tags,
+    # this is set as user_name = str(user.id)
+    # And in the react_comments template tags,
+    # this is set to user.username AFTER the serializer
```

Comment:
I think that is all a bit weird and should be refactored. I mean the code doing that not the test. :)